modules/terraform-aws-ca-lambda/utils/certs/s3.py

```diff
@@ -39,3 +40,58 @@ def s3_upload(
         return s3_upload_file(file, external_s3_bucket_name, key, content_type)
 
     return s3_upload_file(file, internal_s3_bucket_name, key, content_type)
+
+
+def convert_to_json(input_str):
+    # split string by commas
+    pairs = input_str.split(",")
+
+    # split each pair by '=' and construct dictionary
+    json_dictionary = {}
+    for pair in pairs:
+        key, value = pair.split("=")
+        json_dictionary[key] = value
+
+    return json_dictionary
+
+
+def is_cert_gitops(internal_s3_bucket_name, subject):
+    subject_json = convert_to_json(subject)
+
+    cn = subject_json["CN"]
+    o = subject_json.get("O")
+    ou = subject_json.get("OU")
+
+    # get list of GitOps certificates from internal S3 bucket
+    tls_file = s3_download_file(internal_s3_bucket_name, "tls.json")
```

Comment:
If you refactored this method to take `tis_file` as an argument, it would mean you can test the logic without needing to talk to `s3` and could then introduce a unit test for it.